### PR TITLE
boards: shields: add new ci test shield

### PR DESCRIPTION
**Pull request by FRASTM**
This PR is introducing a new shield to test various features like:
-  Digital or Analog I/O,
- PWM and UART loopback,
- SD card on SPI bus
- EEPROM or Temp Sensor on I2C.
    
Especially suitable for testing the SDHC card interface over SPI, with the fat_fs application:
` $ west build -b nucleo_f411re samples/subsys/fs/fat_fs -DSHIELD="ci_shield"`

Signed-off-by: Francois Ramu <francois.ramu@st.com>

### CODE REVIEW

**Comment by FRASTM** [comment]
_Put a FAT-formatted microSD card in the slot and run test application:_

*** Booting Zephyr OS build zephyr-v2.3.0-886-g73ba6c255a7a  ***                                      
[00:00:00.005,000] <inf> spi_ll_stm32: CS control inhibited (no GPIO device)                          
Sector size 512                                                                                       
Memory Size(MB) 1886                                                                                  
Disk mounted.                                                                                         
                                                                                                      
Listing dir /SD: ...                                                                                  
[DIR ] SYSTEM~1                                                                                       
[FILE] TEST.LOG (size = 6144000)                                                                      
[FILE] TEST_B.LOG (size = 2048000)                                                                    
[00:00:00.049,000] <inf> sdhc_spi: Found a ~1886 MiB SDHC card.                                       
[00:00:00.050,000] <inf> sdhc_spi: Manufacturer ID=3 OEM='SD' Name='SU02G' Revision=0x80 Serial=0x6ab6
[00:00:00.050,000] <inf> main: Block count 3862528                                                    
[00:00:00.098,000] <inf> sdhc_spi: Found a ~1886 MiB SDHC card.                                       
[00:00:00.099,000] <inf> sdhc_spi: Manufacturer ID=3 OEM='SD' Name='SU02G' Revision=0x80 Serial=0x6ab6
[00:00:00.099,000] <inf> spi_ll_stm32: CS control inhibited (no GPIO device)

File: boards/shields/ci_shield/boards/nucleo_f411re.defconfig [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+if BOARD_NUCLEO_F411RE
+
+if SPI
+
+config SPI_1
+	default y
+
+endif # SPI
+
+endif # BOARD_NUCLEO_F411RE
```
Comment (erwango) [inline boards/shields/ci_shield/boards/nucleo_f411re.defconfig:14]: This file is not needed

File: boards/shields/ci_shield/boards/nucleo_f411re.overlay [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+ */
+
+&spi1 {
+        status = "okay";
+        cs-gpios = <&gpiob 6 GPIO_ACTIVE_HIGH>;
+
+        sdhc0: sdhc@0 {
+                status = "okay";
+        };
+};
```
Comment (erwango) [inline boards/shields/ci_shield/boards/nucleo_f411re.overlay:14]: This file should not be needed

File: boards/shields/ci_shield/Kconfig.defconfig [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+if SHIELD_CI_SHIELD
+
+config SPI
+	default y
+
+config DISK_ACCESS_SPI_SDHC
+	default y
+
+config DISK_ACCESS_SDHC
+	default y
```
Comment (erwango) [inline boards/shields/ci_shield/Kconfig.defconfig:13]: We want application to control activation, so it should be:
```
if DISK_ACCESS_SDHC

config SPI
	default y

config DISK_ACCESS_SPI_SDHC
	default y

endif
```

**Review by erwango — CHANGES_REQUESTED** [review]

**Comment by pabigot** [comment]
Where can somebody get [one of these](https://os.mbed.com/components/CI-Test-Shield-Minewtech/)?  It's not referenced anywhere on Minew's site anymore; https://github.com/ARMmbed/mbed-HDK is outdated; and I can't find anybody via Google that can ship one.

IIRC there is a plan to produce something like this for Zephyr, though maybe that's been abandoned.

**Comment by jfischer-phytec-iot** [comment]
> IIRC there is a plan to produce something like this for Zephyr, though maybe that's been abandoned.

Yes, I resumed my work on it.

File: boards/shields/ci_shield/Kconfig.defconfig (hunk 3 above)
Comment (FRASTM) [reply boards/shields/ci_shield/Kconfig.defconfig:13]: The application (prj.conf) is configuring CONFIG_DISK_ACCESS=y
So could the `boards/shields/ci_shield/Kconfig.defconfig`  configure with depends on DISK_ACCESS

**Comment by FRASTM** [comment]
- removing boards/shields/ci_shield/boards/xxx

- boards/shields/ci_shield/Kconfig.defconfig depending on DISK_ACCESS

File: boards/shields/ci_shield/boards/nucleo_f411re.overlay (hunk 2 above)
Comment (FRASTM) [reply boards/shields/ci_shield/boards/nucleo_f411re.overlay]: ok

**Comment by erwango** [comment]
> Where can somebody get one of these? It's not referenced anywhere on Minew's site anymore; https://github.com/ARMmbed/mbed-HDK is outdated; and I can't find anybody via Google that can ship one.

That's a good point. We have a few ones internally. I'm not sure how people can get those now. Since schematics are open source, potentially anyone could make some more (recent version is available here: https://github.com/ARMmbed/mbed-HDK-Eagle-Projects/tree/master/Shields/CITestShield). But, as you mention, this doesn't mean people can get them easily.
Based on this, should we consider it can't go upstream, since not publicly available ?

To my understanding, this would be the exact same issue for a potential Zephyr CI test shield. @jfischer-phytec-iot do you confirm?

These shields have definitely an interest for developers, so despite the fact that they can hardly be distributed for now, i'd see an interest in having them available, maybe as a kind of "bootstrap" to show this kind of thing has interest and some maker start producing and distributing it...

**Comment by jfischer-phytec-iot** [comment]
> To my understanding, this would be the exact same issue for a potential Zephyr CI test shield. @jfischer-phytec-iot do you confirm?

Yes, it would offer the same set of functionality.

**Comment by erwango** [comment]
> ld offer the same set of functionality.

Sure, but I was rather pointing at the lack of public distribution (at least this was the status the last time we mentioned it).

**Comment by jfischer-phytec-iot** [comment]
> > ld offer the same set of functionality.
> 
> Sure, but I was rather pointing at the lack of public distribution (at least this was the status the last time we mentioned it).

They would be distributed the same way as the reel and link boards, means with the same issues :-)

**Comment by pabigot** [comment]
> Based on this, should we consider it can't go upstream, since not publicly available ?

That was my point, yes.   We do not have a clear policy regarding the addition or removal of support for hardware that is obsolete, cannot be purchased, or lacks publicly available documentation.  #20548 for example.

**Review by galak — COMMENTED** [review]
Just curious if you know if these shields are something one can buy or not?

**Comment by erwango** [comment]
> Just curious if you know if these shields are something one can buy or not?

@galak, we got made the ones we have, but there is no distribution channel from what I know.

**Review by jfischer-no — APPROVED** [review]

**Comment by FRASTM** [comment]
rebase

**Comment by gmarull** [comment]
FYI I've created a Zephyr clone of the ARM mbed test shield, haven't produced the board yet so there's no guarantee that current design will work. https://github.com/teslabs/zephyr-ci-shield cc: @FRASTM @galak @erwango @pabigot

**Pull request closed without merging**